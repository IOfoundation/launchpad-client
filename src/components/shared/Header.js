import React, {Fragment} from 'react';
import {PropTypes} from 'prop-types';
import {Link, IndexLink} from 'react-router';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';

import BarsIcon from '@Shared/barsIcon';

import * as blogsActions from '@Actions/blogs';
import * as userActions from '@Actions/user';
import {getAuthorization} from '@Utils';

const activeStyles = {opacity: '1'};

const Header = props => {
  let button;

  const setDefaultCategory = () => {
    props.actions.setCategory('front page');
  };

  const signOut = event => {
    const {user, Authorization, router} = props;

    event.preventDefault();
    user.signOut(Authorization, true);
    router.push('/admin-login');
  };

  if (props.isAuth) {
    button = (
      <Fragment>
        <a className="header_link white-link" onClick={signOut}>
          {'Logout'}
        </a>
        <Link className="header_link white-link" to="/admin/profile">
          {'Admin'}
        </Link>
      </Fragment>
    );
  } else {
    button = (
      <Link
        activeStyle={activeStyles}
        className="header_link white-link"
        to="/admin-login"
      >
        {'Admin Login'}
      </Link>
    );
  }

  return (
    <header className="headerContainer" htmlFor="header-dropdown">
      <div className="contentContainer header grid between-xs middle-xs text-semi">
        <div className="header_links-title">
          <span className="header_title">
            <IndexLink
              to="/"
              className="white-link text-bold"
              style={activeStyles}
            >
              {'Welcome to Resource Finder'}
            </IndexLink>
          </span>
        </div>
        <input id="header-dropdown" type="checkbox" name="dropdown" />
        <label htmlFor="header-dropdown" className="collapse-menu-icon">
          <BarsIcon size={24} />
        </label>
        <nav className="header_links-contact text-bold">
          <Link
            activeStyle={activeStyles}
            className="header_link white-link "
            to="/businesses"
          >
            {'Resources'}
          </Link>
          <Link
            activeStyle={activeStyles}
            className="header_link white-link"
            to="/events"
          >
            {'Events'}
          </Link>
          <Link
            activeStyle={activeStyles}
            className="header_link white-link"
            to="/blog"
            onClick={setDefaultCategory}
          >
            {'Blog'}
          </Link>
          {button}
        </nav>
      </div>
    </header>
  );
};

const mapStateToProps = _state => {
  const Authorization = getAuthorization(_state);

  return {
    isAuth: Authorization !== '',
    Authorization,
  };
};

const mapDispatchToProps = _dispatch => {
  return {
    actions: bindActionCreators(blogsActions, _dispatch),
    user: bindActionCreators(userActions, _dispatch),
  };
};

Header.propTypes = {
  actions: PropTypes.shape({
    setCategory: PropTypes.func,
  }),
  Authorization: PropTypes.string,
  homePage: PropTypes.bool.isRequired,
  isAuth: PropTypes.bool,
  router: PropTypes.shape({
    push: PropTypes.func,
  }),
  user: PropTypes.shape({
    signOut: PropTypes.func,
  }),
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Header);
