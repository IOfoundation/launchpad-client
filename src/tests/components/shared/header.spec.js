import React from 'react';
import {mount} from 'enzyme';
import {Link} from 'react-router';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import thunk from 'redux-thunk';
import toJson from 'enzyme-to-json';

import Header from '../../../components/shared/Header';

const initialState = {
  user: {
    Authorization: 'auth',
  },
};
const props = {
  homePage: false,
};
const activeStyles = {opacity: '1'};
const mockStore = configureStore([thunk]);
let store;
let wrapper;

describe('<Header />', () => {
  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <Header {...props} />
      </Provider>
    );
  });

  it('Render a header tag', () => {
    expect(wrapper.find('header').length).toEqual(1);
  });

  describe('Nav Bar', () => {
    it('Render a nav bar', () => {
      expect(wrapper.find('nav').length).toEqual(1);
    });

    it('Render 5 links in the nav bar', () => {
      expect(wrapper.find(Link).length).toEqual(5);
    });

    it('Must have a link to businesses page', () => {
      expect(
        wrapper.contains(
          <Link
            activeStyle={activeStyles}
            className="header_link white-link "
            to="/businesses"
          >
            {'Resources'}
          </Link>
        )
      ).toBe(true);
    });

    it('Must have a link to events page', () => {
      expect(
        wrapper.contains(
          <Link
            activeStyle={activeStyles}
            className="header_link white-link"
            to="/events"
          >
            {'Events'}
          </Link>
        )
      ).toBe(true);
    });

    it('Must have a link to blog page', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Must have a link to admin-login page', () => {
      expect(
        wrapper.contains(
          <Link
            activeStyle={activeStyles}
            className="header_link white-link"
            to="/admin-login"
          >
            {'Admin Login'}
          </Link>
        )
      ).toBe(true);
    });
  });
});
