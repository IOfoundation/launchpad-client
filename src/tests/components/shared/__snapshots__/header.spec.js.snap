// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Nav Bar Must have a link to blog page 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Header)
    homePage={false}
  >
    <Header
      Authorization=""
      actions={
        Object {
          "getAllPosts": [Function],
          "getCategories": [Function],
          "getFeaturedPost": [Function],
          "getFeaturedPostById": [Function],
          "getPostById": [Function],
          "noResults": [Function],
          "resetPostByID": [Function],
          "setCategory": [Function],
        }
      }
      homePage={false}
      isAuth={false}
      user={
        Object {
          "deleteAccount": [Function],
          "login": [Function],
          "passwordRecovery": [Function],
          "resetError": [Function],
          "signOut": [Function],
          "singUp": [Function],
          "updatePasswordAndInformation": [Function],
          "updateUserInformation": [Function],
        }
      }
    >
      <header
        className="headerContainer"
        htmlFor="header-dropdown"
      >
        <div
          className="contentContainer header grid between-xs middle-xs text-semi"
        >
          <div
            className="header_links-title"
          >
            <span
              className="header_title"
            >
              <IndexLink
                activeStyle={
                  Object {
                    "opacity": "1",
                  }
                }
                className="white-link text-bold"
                to="/"
              >
                <Link
                  activeStyle={
                    Object {
                      "opacity": "1",
                    }
                  }
                  className="white-link text-bold"
                  onlyActiveOnIndex={true}
                  style={Object {}}
                  to="/"
                >
                  <a
                    className="white-link text-bold"
                    onClick={[Function]}
                    style={Object {}}
                  >
                    Welcome to Resource Finder
                  </a>
                </Link>
              </IndexLink>
            </span>
          </div>
          <input
            id="header-dropdown"
            name="dropdown"
            type="checkbox"
          />
          <label
            className="collapse-menu-icon"
            htmlFor="header-dropdown"
          >
            <BarsIcon
              size={24}
            >
              <svg
                fill="currentColor"
                height={24}
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 40 40"
                width={24}
              >
                <g>
                  <path
                    d="m37.3 30v2.9q0 0.5-0.4 1t-1 0.4h-31.5q-0.6 0-1-0.4t-0.4-1v-2.9q0-0.6 0.4-1t1-0.4h31.5q0.5 0 1 0.4t0.4 1z m0-11.4v2.8q0 0.6-0.4 1t-1 0.5h-31.5q-0.6 0-1-0.5t-0.4-1v-2.8q0-0.6 0.4-1t1-0.5h31.5q0.5 0 1 0.5t0.4 1z m0-11.5v2.9q0 0.6-0.4 1t-1 0.4h-31.5q-0.6 0-1-0.4t-0.4-1v-2.9q0-0.5 0.4-1t1-0.4h31.5q0.5 0 1 0.4t0.4 1z"
                  />
                </g>
              </svg>
            </BarsIcon>
          </label>
          <nav
            className="header_links-contact text-bold"
          >
            <Link
              activeStyle={
                Object {
                  "opacity": "1",
                }
              }
              className="header_link white-link "
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/businesses"
            >
              <a
                className="header_link white-link "
                onClick={[Function]}
                style={Object {}}
              >
                Resources
              </a>
            </Link>
            <Link
              activeStyle={
                Object {
                  "opacity": "1",
                }
              }
              className="header_link white-link"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/events"
            >
              <a
                className="header_link white-link"
                onClick={[Function]}
                style={Object {}}
              >
                Events
              </a>
            </Link>
            <Link
              activeStyle={
                Object {
                  "opacity": "1",
                }
              }
              className="header_link white-link"
              onClick={[Function]}
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/blog"
            >
              <a
                className="header_link white-link"
                onClick={[Function]}
                style={Object {}}
              >
                Blog
              </a>
            </Link>
            <Link
              activeStyle={
                Object {
                  "opacity": "1",
                }
              }
              className="header_link white-link"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/admin-login"
            >
              <a
                className="header_link white-link"
                onClick={[Function]}
                style={Object {}}
              >
                Admin Login
              </a>
            </Link>
          </nav>
        </div>
      </header>
    </Header>
  </Connect(Header)>
</Provider>
`;
